### ISSUE TITLE
Wrong date & time

### ISSUE THREAD

**Issue by slangpetter** [jomjol/AI-on-the-edge-device#586]
Hi!

I have problems with date & time. 
Everything works as it should except that it thinks it is 1970-01-01. 
Whether I have Timeserver on or off. 
How can i fix this?

Best regards
Petter

<img width="524" alt="Bild2" src="https://user-images.githubusercontent.com/59067063/150142089-b1ee3f50-9d4e-4711-ad37-6d3758d919ba.png">
<img width="275" alt="Bild3" src="https://user-images.githubusercontent.com/59067063/150142096-37dd6203-4d07-4de2-acc1-b87be0aec635.png">
<img width="511" alt="Bild4" src="https://user-images.githubusercontent.com/59067063/150142097-0ee06e5b-1fd1-4672-b1e9-9c1852f8ba06.png">
<img width="590" alt="Bild1" src="https://user-images.githubusercontent.com/59067063/150142084-db132f84-a228-4780-8bca-fbe04da02bd2.png">

**Comment by zuzu59** [comment]
Hello,
I have exactly the same problem !
At each reboot the date is found in 1970, as if the time server was not taken into account.
With version 10.0.1 I did not have this problem!
Thank you
@+

**Comment by stonemaan** [comment]
same here

**Comment by RoberWi** [comment]
Same here. Went back to version 9.2 and the problems are gone.

**Comment by stonemaan** [comment]
> Same here. Went back to version 9.2 and the problems are gone.

unfortunately not for me. I tried all versions since 9.2 - none of them worked. Flashing and OTA did not help. set another time server did not work as well. MQTT does not work anymore...

**Comment by Kalli01** [comment]
Maybe the entry in the config.ini has changed by mistake due to an update.
For me, the following is entered directly in config.ini for Germany.

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = undefined
;AutoAdjustSummertime = false
Hostname = wasser_stadt
;SetupMode = false

**Comment by stonemaan** [comment]
> Maybe the entry in the config.ini has changed by mistake due to an update. For me, the following is entered directly in config.ini for Germany.
> 
> [System] TimeZone = CET-1CEST,M3.5.0,M10.5.0/3 ;TimeServer = undefined ;AutoAdjustSummertime = false Hostname = wasser_stadt ;SetupMode = false

Did not help :/

**Comment by friedpa** [comment]
I am using 9.2.0 with:

[System]
TimeZone = CEST,M3.5.0,M10.5.0/3
TimeServer = fritz.box
Hostname = Gaszähler-ESP32CAM
;SetupMode = false

works fine.

**Comment by stonemaan** [comment]
After erasing flash and re flashing again using FAT32 instead of FAT (do not know if this had any effect) and removing the standardgateway - both of my ESP32 are back and works perfectly.

**Comment by jomjol** [comment]
@slangpetter @Kalli01 : can you test the solution of @stonemaan ?

Otherwise I have no idea!?!

**Comment by Kalli01** [comment]
That was perhaps not clearly written by me.
With me there were no problems with the time.

**Comment by Feinfinger380** [comment]
Same Problem, tried out with some ESP.

Remove the fix IP from the Config solved the Problem.

**Comment by barthm1** [comment]
I have the same problem and I'm not able to correct it - no reaction on cold boot, usrinf a ntp server
![screengas](https://user-images.githubusercontent.com/29302671/150675205-59ee1697-e002-4f5d-8099-73cfa3f5458a.JPG)
The wrong time belongs to the next day
![screengas2](https://user-images.githubusercontent.com/29302671/150675243-49365634-e199-4b07-89dd-6f06fc9be3fb.JPG)
![screengas3](https://user-images.githubusercontent.com/29302671/150675270-9b61a1d6-6504-4069-ae30-2c053411aff6.JPG)

**Comment by Feinfinger380** [comment]
> I have the same problem and I'm not able to correct it - no reaction on cold boot, using a ntp server
> 
> ![screengas](https://user-images.githubusercontent.com/29302671/150675205-59ee1697-e002-4f5d-8099-73cfa3f5458a.JPG) The wrong time belongs to the next day ![screengas2](https://user-images.githubusercontent.com/29302671/150675243-49365634-e199-4b07-89dd-6f06fc9be3fb.JPG) ![screengas3](https://user-images.githubusercontent.com/29302671/150675270-9b61a1d6-6504-4069-ae30-2c053411aff6.JPG)

Please show the serial Log at startup.

**Comment by jomjol** [comment]
@barthm1 : sorry I don't get the problem. Can you explain it a bit more in detail?

**Comment by barthm1** [comment]
Datum liegt 2 Tage in der Zukunft Uhrzeit ist auch irgendwo - Durch Kaltstart, eintragen eines NTP Servers kann das nicht korrigiert werden. Löschen der Log-Dateien habe ich auch probiert - Datum Uhrzeit bleibt falsch. Timezone ist korrekt

**Comment by friedpa** [comment]
@barthm1 Bitte einmal den Config.ini Eintrag hier posten.

**Comment by Feinfinger380** [comment]
@barthm1 

Hast du ne feste IP vergeben?

**Comment by barthm1** [comment]
@friedpa warum der hostname "nicht" übertragen wird weiß ich auch nicht


[System]

TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
TimeServer = pool.ntp.org
;AutoAdjustSummertime = 
Hostname = gasmeter
;SetupMode = false

anderer Zähler der funktioniert

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = undefined
;AutoAdjustSummertime = false
;Hostname = undefined
SetupMode = false

**Comment by barthm1** [comment]
Last restart steht auch falsch

![screengas4](https://user-images.githubusercontent.com/29302671/151242689-7100bfa9-1a8b-4610-93d5-27f6d3b2046f.JPG)

**Comment by michaeljoos72** [comment]
An alle mit Zeitproblemen: Versucht einfach mal per OTA eine ältere FW zu flashen --> z.B auf 10.1.0, dann Zeit kontrollieren und falls OK wieder die aktuellste FW flashen. Hat bei mir jetzt schon 2x geholfen.

**Comment by jomjol** [comment]
Zuerstmal zum Ablauf:
1) nach dem Booten wird erstmal sehr früh versucht, die Zeit zu bekommen, damit die Logeinträge stimmen. Da zu dem Zeitpunkt die `config.ini` nicht gelesen wurde, wird ein Standart-Zeitserver verwendet: pool.ntp.org
2) Falls der nicht nach 10 Versuchen erreicht wird, bleibt die Zeit bei 01.01.1970...
3) Wenn in der `config.ini` ein aktiver Timeserver-Eintrag gefunden wird, wird dieser gesetzt und die Zeit dann nochmals dort mit max. 10 Versuchen abgefragt und gesetzt.
4) Die Zeitzone wird über `TimeZone = CET-1CEST,M3.5.0,M10.5.0/3` gesetzt (auch erstmal vorläufig, bis er in der Config.ini ggf. aktualisiert wird.

Ich habe gerade bei mir nochmal verschiedene Konfigurationen mit eigenem und externem Zeitserver getestet. Läuft eigentlich wie erwartet. Mit einer Auffälligkeit: meinen lokalen Timerserver (FritzBox) wird nur erkannt, wenn ich die IP direkt eingebe - mit `fritz.box` geht es nicht.

Wenn die Uhr um 2 Tage falsch geht, kann es nur am Timeserver oder einem Fehler im Format `TimeZone` liegen.

**Comment by barthm1** [comment]
Sobald ich die IP-Aadresse der Fritzbox eIntrage, wird nach einem Neustart die Zeit korrekt gesetzt  - Problem scheint also nur die Namensauflösung zu sein

**Comment by Mad-007** [comment]
Danke für den Tipp - nach direkter Eingabe der IP Adresse der FritzBox wird auch bei mir die Zeit wieder richtig gesetzt.

**Comment by Gunter1710** [comment]
Also die Fritzbox IP-Adresse hat bei mir auch geholfen. Verstehen muss man das nicht aber es geht :-)

**Comment by jomjol** [comment]
Dann schließe ich das jetzt mal.

**Comment by zumheulen** [comment]
@jomjol You "closed" here but didn't generally fix the problem ;-)
I too went almost insane all morning today, although I had set it up perfectly with V12.01 (SD card was defective). Was timestamp 1970!
I tried all possible time servers listed on the net, and rebooted. Some, but not all, I was able to ping successfully beforehand. Since I don't own a "Fritzbox" but a UniFi Dreammachine and everything is a little more complicated, I can't rule out that this is also partially responsible for "problems".
At least it was unsuccessful to give the UniFi its IP because, according to internet research, it probably does not provide a timestamp... (no guarantee)
WHAT, however, brought immediate success (a tip on the Internet), my Synology NAS provides the time within the network and boom, everything worked! :-)
